### PR TITLE
Reduced some scalac warnings.

### PR DESCRIPTION
**Pull request by takayahilton**
- match may not be exhaustive.
- Unused import.
- Invalid implicitNotFound message.
- Adapting argument list.
- method copyArrayToImmutableIndexedSeq in class LowPriorityImplicits2 is deprecated.
- Adding a number and a String is deprecated.

### CODE REVIEW

**Comment by raboof** [comment]
OK TO TEST

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala [hunk 1]
```diff
@@ -138,6 +138,9 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with StageLoggi
         } else pushGOAWAY(Http2Protocol.ErrorCode.PROTOCOL_ERROR, "Got unexpected mid-stream HEADERS frame")
 
         maybeFinishStream(h.endStream)
+
+      case _: ContinuationFrame | _: HeadersFrame | _: PriorityFrame | _: PushPromiseFrame | _: UnknownFrameEvent | _: WindowUpdateFrame =>
+        throw new UnsupportedOperationException(event.frameTypeName + " is unsupported.")
```
Comment (raboof) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala]: Perhaps also log the 'this' class?

File: akka-http/src/main/scala/akka/http/scaladsl/server/directives/SecurityDirectives.scala [hunk 2]
```diff
@@ -314,7 +314,7 @@ object Credentials {
         new Credentials.Provided(token) {
           def verify(secret: String, hasher: String => String): Boolean = secret secure_== hasher(token)
         }
-      case Some(GenericHttpCredentials(scheme, token, params)) =>
+      case Some(_: HttpCredentials) =>
```
Comment (raboof) [inline akka-http/src/main/scala/akka/http/scaladsl/server/directives/SecurityDirectives.scala]: Perhaps log the type of credentials, as we're no longer checking it's actually `GenericHttpCredentials` here?

File: akka-http/src/main/scala/akka/http/javadsl/server/directives/DebuggingDirectives.scala [hunk 3]
```diff
@@ -57,7 +57,7 @@ abstract class DebuggingDirectives extends CookieDirectives {
    * @param level One of the log levels defined in akka.event.Logging
    */
   def logResult(marker: String, level: LogLevel, inner: Supplier[Route]): Route = RouteAdapter {
-    D.logResult(marker, level) { inner.get.delegate }
+    D.logResult(marker -> level) { inner.get.delegate }
```
Comment (raboof) [inline akka-http/src/main/scala/akka/http/javadsl/server/directives/DebuggingDirectives.scala]:
```suggestion
    D.logResult((marker, level)) { inner.get.delegate }
```

File: akka-http/src/main/scala/akka/http/javadsl/server/directives/DebuggingDirectives.scala [hunk 4]
```diff
@@ -34,7 +34,7 @@ abstract class DebuggingDirectives extends CookieDirectives {
    * @param level One of the log levels defined in akka.event.Logging
    */
   def logRequest(marker: String, level: LogLevel, inner: Supplier[Route]): Route = RouteAdapter {
-    D.logRequest(marker, level) { inner.get.delegate }
+    D.logRequest(marker -> level) { inner.get.delegate }
```
Comment (raboof) [inline akka-http/src/main/scala/akka/http/javadsl/server/directives/DebuggingDirectives.scala]: `->` doesn't really make sense here, as it's not a mapping - tuple notation might be clearer?
```suggestion
    D.logRequest((marker, level)) { inner.get.delegate }
```

File: akka-http/src/main/scala/akka/http/scaladsl/server/directives/SecurityDirectives.scala [hunk 5]
```diff
@@ -314,8 +314,8 @@ object Credentials {
         new Credentials.Provided(token) {
           def verify(secret: String, hasher: String => String): Boolean = secret secure_== hasher(token)
         }
-      case Some(GenericHttpCredentials(scheme, token, params)) =>
-        throw new UnsupportedOperationException("cannot verify generic HTTP credentials")
+      case Some(c) =>
+        throw new UnsupportedOperationException(s"Credentials does not support $c.")
```
Comment (jrudolph) [inline akka-http/src/main/scala/akka/http/scaladsl/server/directives/SecurityDirectives.scala]: Printing `c` might leak secrets to the exception message. Let's use `c.scheme` instead.

```suggestion
        throw new UnsupportedOperationException(s"Credentials does not support scheme '${c.scheme}'.")
```

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/Multipart.scala [hunk 6]
```diff
@@ -358,7 +358,10 @@ object Multipart {
     /** INTERNAL API */
     @InternalApi
     private[akka] def createStrict(fields: Map[String, akka.http.javadsl.model.HttpEntity.Strict]): Multipart.FormData.Strict = Multipart.FormData.Strict {
-      fields.iterator.map { case (name, entity: akka.http.scaladsl.model.HttpEntity.Strict) => Multipart.FormData.BodyPart.Strict(name, entity) }.to(scala.collection.immutable.IndexedSeq)
+      fields.iterator.map {
+        case (name, entity: akka.http.scaladsl.model.HttpEntity.Strict) => Multipart.FormData.BodyPart.Strict(name, entity)
+        case _ => throw new UnsupportedOperationException("must strict")
```
Comment (jrudolph) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/Multipart.scala]:
```suggestion
        case _ => throw new IllegalStateException("Entity is expected to be strict")
```

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala [hunk 7]
```diff
@@ -138,6 +138,9 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with StageLoggi
         } else pushGOAWAY(Http2Protocol.ErrorCode.PROTOCOL_ERROR, "Got unexpected mid-stream HEADERS frame")
 
         maybeFinishStream(h.endStream)
+
+      case _: ContinuationFrame | _: HeadersFrame | _: PriorityFrame | _: PushPromiseFrame | _: UnknownFrameEvent | _: WindowUpdateFrame =>
```
Comment (jrudolph) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala]: Can you change this to:

```scala
case x => throw new IllegalStateException(s"Unexpected frame type ${x.frameTypeName} in state ${this.getClass.getName}")
```

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/framing/FrameRenderer.scala [hunk 8]
```diff
@@ -139,6 +139,8 @@ private[http2] object FrameRenderer {
           streamId,
           renderPriorityInfo(frame)
         )
+      case _: ParsedHeadersFrame | _: UnknownFrameEvent =>
```
Comment (jrudolph) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/framing/FrameRenderer.scala]:
```scala
case x => throw new IllegalStateException(s"Unexpected frame type ${x.frameTypeName}")
```

**Comment by jrudolph** [comment]
Thanks for following up, @takayahilton.

**Pull request merged**